### PR TITLE
Renovate proposal

### PR DESCRIPTION
**Pull request by lmilbaum**
## Changes introduced with this PR

Please explain your changes here.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sandrobonazzola** [comment]
Perhaps also add some link to `renovate` documentation

**Comment by lmilbaum** [comment]
> Perhaps also add some link to `renovate` documentation

Done.

**Comment by HonkingGoose** [comment]
I help maintain Renovate bot's documentation, so obviously I like Renovate bot. 😉 

## Why pick Renovate?

Copy/paste from the Renovate docs homepage:

> ### Why use Renovate?
>
> - Get pull requests to update your dependencies and lock files
> - Reduce noise by scheduling when Renovate creates PRs
> - Renovate finds relevant package files automatically, including in monorepos
> - You can customize the bot's behavior with configuration files
> - Share your configuration with ESLint-like config presets
> - Get replacement PRs to migrate from a deprecated dependency to the community suggested replacement (npm packages only)
> - Open source

## Killer features

- Dependency Dashboard to get an overview of all your pending/in-progress updates 
[^dep-dashboard]
- Keep noise down by package grouping, and optionally automerging [^noise-reduction]
- Dependency Dashboard Approval Workflow to only get updates when you select them from the Dashboard [^dependency-dashboard-approval]

Here's the Dependency Dashboard in action:

- https://github.com/renovatebot/renovate/issues/2958

## Management overhead?

> - we need to allow an app
> - someone needs to go register an account with them
> - or self-host

The overhead depends on how you're going to run Renovate. You can:

- use the GitHub-hosted app version of Renovate
- use the "Forking Renovate" app version (needs less permissions) [^permissions]
- self-host

### Easiest way to get started

The easiest way to run the bot is by installing it as a GitHub-hosted app. [^github-hosted-app-installation] You can select what repositories you want Renovate to access after installing the app if you don't want Renovate to run on all your repositories.

[^dependency-dashboard-approval]: https://docs.renovatebot.com/key-concepts/dashboard/#dependency-dashboard-approval-workflow
[^permissions]: https://docs.renovatebot.com/security-and-permissions/
[^forking-renovate-app]: https://docs.renovatebot.com/getting-started/running/#forking-renovate-app
[^dep-dashboard]: https://docs.renovatebot.com/key-concepts/dashboard/
[^noise-reduction]: https://docs.renovatebot.com/noise-reduction/
[^github-hosted-app-installation]: https://docs.renovatebot.com/getting-started/installing-onboarding/#hosted-githubcom-app

File: art-decisions/proposals/2023-03-08-automated-dependency-updates.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+1. Pull request overload: Renovate can generate a large number of pull requests, especially if you have many repositories or dependencies. This can create a lot of noise in your pull request feed and may require additional effort to manage.
+
+2. False positives: Renovate may sometimes generate pull requests for updates that don't actually improve your dependencies or may introduce new issues. This can create additional work and may require manual intervention to resolve.
+
+3. Configurability complexity: While Renovate offers a lot of configuration options, this can also make it more complex to set up and manage than other dependency update tools.
+
+4. Limited control over updates: Renovate automates the update process, but you may not have full control over which updates are applied to your dependencies. This can be a concern if you need to ensure strict compatibility or stability with certain packages or versions.
+
+5. Potential conflicts with other automation: If you have other automation tools or processes in place that also manage dependency updates, Renovate may conflict with them and create additional work or confusion.
```
Comment (HonkingGoose) [inline art-decisions/proposals/2023-03-08-automated-dependency-updates.md:53]: Do _not_ run Dependabot and Renovate together, you'll end up with a mess of open PRs.

If you want to get `vulnerabilityAlerts` [^vulnerabilityAlerts] then you'll need to enable the "Dependency Graph" and "Dependabot Alerts". You still keep the "normal Dependabot" _disabled_ though!

[^vulnerabilityAlerts]: https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts

**Review by HonkingGoose — COMMENTED** [review]
Some comments on the proposal text.

**Comment by dustinblack** [comment]
As discussed in today's call, I'm flipping this back to draft for the time being because we have some ART administrative things we should take care of before we properly consider other proposals.

**Review by Harshith-umesh — APPROVED** [review]

**Review by jaredoconnell — COMMENTED** [review]
For the currently maintained projects, this seems to me to be extra work.

I think this would make sense as an option for projects, but not the default. But that could lead to more fragmentation.

This would make more sense if we switched off of GitHub.

**Review by mfleader — APPROVED** [review]

**Comment by lmilbaum** [comment]
> For the currently maintained projects, this seems to me to be extra work.
> 
> I think this would make sense as an option for projects, but not the default. But that could lead to more fragmentation.
> 
> This would make more sense if we switched off of GitHub.

I understand your concern. If that may help, my team (platform engineering) will own the on-boarding process and maintenance.

**Comment by dustinblack** [comment]
Per today's discussion, we are going to have a look at this functionality as a prototype using a repo or two from the Arcalot org. Seeing directly how this change affects our maintenance workflows and helps us will be useful to better inform the decision here.

**Comment by dustinblack** [comment]
We have implemented a prototype of this implementation with one plugin repo that has open dependabot PRs so that we can see a live comparison against Renovate. I invite the Round Table members to please have a look at the open PRs for this repo to see what Renovate offers us:

https://github.com/arcalot/arcaflow-plugin-aws-ec2-control/pulls

Implementing this required allowing the platform-engineering-bot write access to the repo, and a Renovate configuration file to be added:

https://github.com/arcalot/arcaflow-plugin-aws-ec2-control/blob/main/renovate.json

So the impact for implementation across the org is low, and we can do it in a phased way if this proposal is approved. I believe we would grant the platform-engineering-bot write access at the org level, and then add this config to each repo for which we want to enable Renovate.

Based on the conversations and this update, **I am resetting the votes** on this proposal and adjusting the voting period to **14 days**, targeted to end on **Wednesday, July 19,** or when quorum is reached, whichever comes first.

**Comment by jdowni000** [comment]
I would def like to see this with all the build issues fixed manually before dependabot gives this a go. It seems we have had different breaks within the last week or so.

**Review by jaredoconnell — APPROVED** [review]

**Review by Harshith-umesh — APPROVED** [review]

**Review by jdowni000 — APPROVED** [review]

**Review by dustinblack — APPROVED** [review]

**Review by tsebastiani — APPROVED** [review]

**Review by ozanunsal — APPROVED** [review]

**Review by sandrobonazzola — APPROVED** [review]

**Comment by dustinblack** [comment]
Merging as approved with a quorum of votes from the ART

**Pull request merged**